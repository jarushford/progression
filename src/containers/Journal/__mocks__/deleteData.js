export const deleteDataThunk = jest.fn()