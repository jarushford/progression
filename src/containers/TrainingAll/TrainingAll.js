import React from 'react'
import { connect }  from 'react-redux';
import { withRouter } from 'react-router-dom'
import { toggleComplete } from '../../actions'
import { uid } from 'react-uid'
import '../../main.scss'

function TrainingAll({ trainingData, toggleComplete }) {
  const trainingKeys = Object.keys(trainingData).sort()
  const trainingRender = trainingKeys.map(workout => {
    const key = uid(trainingData[workout])
    return (
        <article className={`project ${trainingData[workout].completed && 'completed'}`} key={key} onClick={() => toggleComplete(workout)}>
          <div>
            <h2 className="project-name">{workout}</h2>
            <h5 className="project-grade">{trainingData[workout].type}</h5>
          </div>
          <h5 className="project-location">{trainingData[workout].description}</h5>
        </article>
    )
  })

  return (
    <section className="projects workout">
      <div className="projects-header workout">
        <h1 className="projects-title workout">ALL WORKOUTS</h1>
        <button className="add-workout">Add Workout</button>
      </div>
      {trainingRender}
    </section>
  )
}

const mapStateToProps = (state) => ({
  trainingData: state.trainingData
})

const mapDispatchToProps = (dispatch) => ({
  toggleComplete: (key) => dispatch(toggleComplete(key))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(TrainingAll))