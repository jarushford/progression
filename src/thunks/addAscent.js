import { fetchAscentsThunk } from './fetchAscents'
import { clearAscents, setError } from '../actions'

export const addAscentThunk = (ascent) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/ascents/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(ascent),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add ascent')
      }
      await dispatch(clearAscents())
      dispatch(fetchAscentsThunk(ascent.user_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}