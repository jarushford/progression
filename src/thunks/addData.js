import { clearThunkHelper } from '../utils/thunkHelpers'
import { fetchDataThunk } from './fetchData'
import { setError } from '../actions'

export const addDataThunk = (data, type) => {
  return async (dispatch) => {
    try {
      const url = `http://localhost:3000/api/progressionusers/${type}s/new`
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json"
        }
      })

      if (!response.ok) {
        throw Error(`Could not add ${type}`)
      }

      await dispatch(clearThunkHelper(type)())
      type === 'journal' || type === 'milestone'
        ? dispatch(fetchDataThunk(data.user_id, data.project_id, type))
        : dispatch(fetchDataThunk(data.user_id, '', type))

      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}