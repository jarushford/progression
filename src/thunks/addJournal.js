import { fetchJournalThunk } from './fetchJournal'
import { clearJournal, setError } from '../actions'

export const addJournalThunk = (entry) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/journal/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(entry),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add entry')
      }
      await dispatch(clearJournal())
      dispatch(fetchJournalThunk(entry.user_id, entry.project_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}