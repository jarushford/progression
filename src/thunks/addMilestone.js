import { fetchMilestonesThunk } from './fetchMilestones'
import { clearMilestones, setError } from '../actions'

export const addMilestoneThunk = (milestone) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/milestones/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(milestone),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add project')
      }
      await dispatch(clearMilestones())
      dispatch(fetchMilestonesThunk(milestone.user_id, milestone.project_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}