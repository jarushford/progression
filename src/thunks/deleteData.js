import { clearThunkHelper } from '../utils/thunkHelpers'
import { fetchDataThunk } from './fetchData'
import { setError } from '../actions'

export const deleteDataThunk = (item_id, user_id, project_id, type) => {
  return async (dispatch) => {
    try {
      const data = {item_id, user_id, project_id}
      let url
      type === 'journal' || type === 'milestone'
        ? url = `http://localhost:3000/api/progressionusers/${user_id}/projects/${project_id}/${type}s/${item_id}`
        : url = `http://localhost:3000/api/progressionusers/${user_id}/${type}s/${item_id}`

      const response = await fetch(url, {
        method: 'DELETE',
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json"
        }
      })

      if (!response.ok) {
        throw Error(`Could not remove ${type}`)
      }

      await dispatch(clearThunkHelper(type)())

      type === 'journal' || type === 'milestone'
        ? dispatch(fetchDataThunk(data.user_id, data.project_id, type))
        : dispatch(fetchDataThunk(data.user_id, '', type))

    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}