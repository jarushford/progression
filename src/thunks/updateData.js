import { fetchDataThunk } from './fetchData'
import { setError } from '../actions'

export const updateDataThunk = (data, type) => {
  return async (dispatch) => {
    try {
      if (type === 'workout') {
        data.completed = !data.completed
      }

      const url = `https://progression-backend.herokuapp.com/api/progressionusers/${data.user_id}/${type}s/${data.id}`
      const response = await fetch(url, {
        method: 'PUT',
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json"
        }
      })

      if (!response.ok) {
        throw Error(`Could not update ${type}`)
      }

      await dispatch(fetchDataThunk(data.user_id, '', type))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}