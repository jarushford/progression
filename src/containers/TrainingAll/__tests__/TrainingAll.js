import React from 'react'
import { shallow } from 'enzyme'
import { deleteDataThunk } from '../../../thunks/deleteData'
import { mapStateToProps, mapDispatchToProps, TrainingAll } from '../TrainingAll'
import { updateDataThunk } from '../../../thunks/updateData'

jest.mock('../../../thunks/updateData')
jest.mock('../../../thunks/deleteData')

describe('Training All', () => {
  let mockUser 
  let mockData
  let mockToggle
  let mockDelete

  beforeEach(() => {
    mockUser = {
      name: 'Larry',
      email: 'larry@gmail.com',
      password: 'chrimbus',
      id: 8
    }
    mockData = {
      0: { workout_date: '3/3/13', type: 'Power', description: 'do stuff', completed: true },
      1: { workout_date: '3/1/13', type: 'Power', description: 'do other stuff', completed: true },
      2: { workout_date: '3/6/13', type: 'Power', description: 'do other stuff', completed: true },
      3: { workout_date: '3/7/13', type: 'Power', description: 'do other stuff', completed: true }
    }
    mockToggle = jest.fn()
    mockDelete = jest.fn()
  })

  describe('TrainingAll Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<TrainingAll user={mockUser} trainingDataUnclean={mockData} toggleComplete={mockToggle} deleteWorkout={mockDelete} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no user', () => {
      const noUser = { name: '', email: '', password: '' }
      const wrapper = shallow(<TrainingAll user={noUser} trainingDataUnclean={mockData} toggleComplete={mockToggle} deleteWorkout={mockDelete} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no data', () => {
      const noData = {}
      const wrapper = shallow(<TrainingAll user={mockUser} trainingDataUnclean={noData} toggleComplete={mockToggle} deleteWorkout={mockDelete} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should toggle a workout completed', () => {
      const wrapper = shallow(<TrainingAll user={mockUser} trainingDataUnclean={mockData} toggleComplete={mockToggle} deleteWorkout={mockDelete} />)

      wrapper.find('.project').first().simulate('click')

      expect(mockToggle).toBeCalled()
    })

    it('should delete a workout on click', () => {
      const wrapper = shallow(<TrainingAll user={mockUser} trainingDataUnclean={mockData} deleteWorkout={mockDelete} toggleComplete={mockToggle} />)

      wrapper.find('.fa-times').first().simulate('click')

      expect(mockDelete).toBeCalled()
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = { currentUser: mockUser, trainingData: mockData }
      const expected = { user: mockUser, trainingDataUnclean: mockData }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method toggleComplete', () => {
      const mockDispatch = jest.fn()
      const expected = updateDataThunk()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleComplete()

      expect(mockDispatch).toBeCalledWith(expected)
    })

    it('should return a props object with a method deleteWorkout', () => {
      const mockDispatch = jest.fn()
      const expected = deleteDataThunk()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.deleteWorkout()

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })
})