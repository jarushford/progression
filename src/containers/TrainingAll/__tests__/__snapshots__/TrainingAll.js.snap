// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training All TrainingAll Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainingAll
    deleteWorkout={[MockFunction]}
    toggleComplete={[MockFunction]}
    trainingDataUnclean={
      Object {
        "0": Object {
          "completed": true,
          "description": "do stuff",
          "type": "Power",
          "workout_date": "3/3/31",
        },
      }
    }
    user={
      Object {
        "email": "larry@gmail.com",
        "id": 8,
        "name": "Larry",
        "password": "chrimbus",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="projects-header workout"
        >
          <h1
            className="projects-title workout"
          >
            ALL WORKOUTS
          </h1>
          <div
            className="buttons"
          >
            <Link
              replace={false}
              to="/training"
            >
              <button
                className="view-workouts"
              >
                Back To Calendar
              </button>
            </Link>
            <Link
              replace={false}
              to="/training/add"
            >
              <button
                className="add-workout"
              >
                Add Workout
              </button>
            </Link>
          </div>
        </div>,
        Array [
          <article
            className="project completed"
            onClick={[Function]}
          >
            <div>
              <h2
                className="project-name"
              >
                3/3/31
              </h2>
              <h5
                className="project-grade"
              >
                Power
              </h5>
            </div>
            <h5
              className="project-location"
            >
              do stuff
            </h5>
            <i
              className="fas fa-times"
              onClick={[Function]}
            />
          </article>,
        ],
      ],
      "className": "projects workout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="projects-title workout"
            >
              ALL WORKOUTS
            </h1>,
            <div
              className="buttons"
            >
              <Link
                replace={false}
                to="/training"
              >
                <button
                  className="view-workouts"
                >
                  Back To Calendar
                </button>
              </Link>
              <Link
                replace={false}
                to="/training/add"
              >
                <button
                  className="add-workout"
                >
                  Add Workout
                </button>
              </Link>
            </div>,
          ],
          "className": "projects-header workout",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ALL WORKOUTS",
              "className": "projects-title workout",
            },
            "ref": null,
            "rendered": "ALL WORKOUTS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/training"
                >
                  <button
                    className="view-workouts"
                  >
                    Back To Calendar
                  </button>
                </Link>,
                <Link
                  replace={false}
                  to="/training/add"
                >
                  <button
                    className="add-workout"
                  >
                    Add Workout
                  </button>
                </Link>,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="view-workouts"
                  >
                    Back To Calendar
                  </button>,
                  "replace": false,
                  "to": "/training",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Back To Calendar",
                    "className": "view-workouts",
                  },
                  "ref": null,
                  "rendered": "Back To Calendar",
                  "type": "button",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="add-workout"
                  >
                    Add Workout
                  </button>,
                  "replace": false,
                  "to": "/training/add",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Workout",
                    "className": "add-workout",
                  },
                  "ref": null,
                  "rendered": "Add Workout",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "uid1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h2
                className="project-name"
              >
                3/3/31
              </h2>
              <h5
                className="project-grade"
              >
                Power
              </h5>
            </div>,
            <h5
              className="project-location"
            >
              do stuff
            </h5>,
            <i
              className="fas fa-times"
              onClick={[Function]}
            />,
          ],
          "className": "project completed",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="project-name"
                >
                  3/3/31
                </h2>,
                <h5
                  className="project-grade"
                >
                  Power
                </h5>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "3/3/31",
                  "className": "project-name",
                },
                "ref": null,
                "rendered": "3/3/31",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Power",
                  "className": "project-grade",
                },
                "ref": null,
                "rendered": "Power",
                "type": "h5",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "do stuff",
              "className": "project-location",
            },
            "ref": null,
            "rendered": "do stuff",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-times",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="projects-header workout"
          >
            <h1
              className="projects-title workout"
            >
              ALL WORKOUTS
            </h1>
            <div
              className="buttons"
            >
              <Link
                replace={false}
                to="/training"
              >
                <button
                  className="view-workouts"
                >
                  Back To Calendar
                </button>
              </Link>
              <Link
                replace={false}
                to="/training/add"
              >
                <button
                  className="add-workout"
                >
                  Add Workout
                </button>
              </Link>
            </div>
          </div>,
          Array [
            <article
              className="project completed"
              onClick={[Function]}
            >
              <div>
                <h2
                  className="project-name"
                >
                  3/3/31
                </h2>
                <h5
                  className="project-grade"
                >
                  Power
                </h5>
              </div>
              <h5
                className="project-location"
              >
                do stuff
              </h5>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </article>,
          ],
        ],
        "className": "projects workout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="projects-title workout"
              >
                ALL WORKOUTS
              </h1>,
              <div
                className="buttons"
              >
                <Link
                  replace={false}
                  to="/training"
                >
                  <button
                    className="view-workouts"
                  >
                    Back To Calendar
                  </button>
                </Link>
                <Link
                  replace={false}
                  to="/training/add"
                >
                  <button
                    className="add-workout"
                  >
                    Add Workout
                  </button>
                </Link>
              </div>,
            ],
            "className": "projects-header workout",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ALL WORKOUTS",
                "className": "projects-title workout",
              },
              "ref": null,
              "rendered": "ALL WORKOUTS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/training"
                  >
                    <button
                      className="view-workouts"
                    >
                      Back To Calendar
                    </button>
                  </Link>,
                  <Link
                    replace={false}
                    to="/training/add"
                  >
                    <button
                      className="add-workout"
                    >
                      Add Workout
                    </button>
                  </Link>,
                ],
                "className": "buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="view-workouts"
                    >
                      Back To Calendar
                    </button>,
                    "replace": false,
                    "to": "/training",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Back To Calendar",
                      "className": "view-workouts",
                    },
                    "ref": null,
                    "rendered": "Back To Calendar",
                    "type": "button",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="add-workout"
                    >
                      Add Workout
                    </button>,
                    "replace": false,
                    "to": "/training/add",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Add Workout",
                      "className": "add-workout",
                    },
                    "ref": null,
                    "rendered": "Add Workout",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h2
                  className="project-name"
                >
                  3/3/31
                </h2>
                <h5
                  className="project-grade"
                >
                  Power
                </h5>
              </div>,
              <h5
                className="project-location"
              >
                do stuff
              </h5>,
              <i
                className="fas fa-times"
                onClick={[Function]}
              />,
            ],
            "className": "project completed",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="project-name"
                  >
                    3/3/31
                  </h2>,
                  <h5
                    className="project-grade"
                  >
                    Power
                  </h5>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "3/3/31",
                    "className": "project-name",
                  },
                  "ref": null,
                  "rendered": "3/3/31",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Power",
                    "className": "project-grade",
                  },
                  "ref": null,
                  "rendered": "Power",
                  "type": "h5",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "do stuff",
                "className": "project-location",
              },
              "ref": null,
              "rendered": "do stuff",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-times",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Training All TrainingAll Component should match the snapshot with no data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainingAll
    deleteWorkout={[MockFunction]}
    toggleComplete={[MockFunction]}
    trainingDataUnclean={Object {}}
    user={
      Object {
        "email": "larry@gmail.com",
        "id": 8,
        "name": "Larry",
        "password": "chrimbus",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="projects-header workout"
        >
          <h1
            className="projects-title workout"
          >
            ALL WORKOUTS
          </h1>
          <div
            className="buttons"
          >
            <Link
              replace={false}
              to="/training"
            >
              <button
                className="view-workouts"
              >
                Back To Calendar
              </button>
            </Link>
            <Link
              replace={false}
              to="/training/add"
            >
              <button
                className="add-workout"
              >
                Add Workout
              </button>
            </Link>
          </div>
        </div>,
        <h1
          className="no-data-msg"
        >
          You have no saved workouts
        </h1>,
      ],
      "className": "projects workout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="projects-title workout"
            >
              ALL WORKOUTS
            </h1>,
            <div
              className="buttons"
            >
              <Link
                replace={false}
                to="/training"
              >
                <button
                  className="view-workouts"
                >
                  Back To Calendar
                </button>
              </Link>
              <Link
                replace={false}
                to="/training/add"
              >
                <button
                  className="add-workout"
                >
                  Add Workout
                </button>
              </Link>
            </div>,
          ],
          "className": "projects-header workout",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ALL WORKOUTS",
              "className": "projects-title workout",
            },
            "ref": null,
            "rendered": "ALL WORKOUTS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/training"
                >
                  <button
                    className="view-workouts"
                  >
                    Back To Calendar
                  </button>
                </Link>,
                <Link
                  replace={false}
                  to="/training/add"
                >
                  <button
                    className="add-workout"
                  >
                    Add Workout
                  </button>
                </Link>,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="view-workouts"
                  >
                    Back To Calendar
                  </button>,
                  "replace": false,
                  "to": "/training",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Back To Calendar",
                    "className": "view-workouts",
                  },
                  "ref": null,
                  "rendered": "Back To Calendar",
                  "type": "button",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="add-workout"
                  >
                    Add Workout
                  </button>,
                  "replace": false,
                  "to": "/training/add",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Workout",
                    "className": "add-workout",
                  },
                  "ref": null,
                  "rendered": "Add Workout",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have no saved workouts",
          "className": "no-data-msg",
        },
        "ref": null,
        "rendered": "You have no saved workouts",
        "type": "h1",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="projects-header workout"
          >
            <h1
              className="projects-title workout"
            >
              ALL WORKOUTS
            </h1>
            <div
              className="buttons"
            >
              <Link
                replace={false}
                to="/training"
              >
                <button
                  className="view-workouts"
                >
                  Back To Calendar
                </button>
              </Link>
              <Link
                replace={false}
                to="/training/add"
              >
                <button
                  className="add-workout"
                >
                  Add Workout
                </button>
              </Link>
            </div>
          </div>,
          <h1
            className="no-data-msg"
          >
            You have no saved workouts
          </h1>,
        ],
        "className": "projects workout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="projects-title workout"
              >
                ALL WORKOUTS
              </h1>,
              <div
                className="buttons"
              >
                <Link
                  replace={false}
                  to="/training"
                >
                  <button
                    className="view-workouts"
                  >
                    Back To Calendar
                  </button>
                </Link>
                <Link
                  replace={false}
                  to="/training/add"
                >
                  <button
                    className="add-workout"
                  >
                    Add Workout
                  </button>
                </Link>
              </div>,
            ],
            "className": "projects-header workout",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ALL WORKOUTS",
                "className": "projects-title workout",
              },
              "ref": null,
              "rendered": "ALL WORKOUTS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/training"
                  >
                    <button
                      className="view-workouts"
                    >
                      Back To Calendar
                    </button>
                  </Link>,
                  <Link
                    replace={false}
                    to="/training/add"
                  >
                    <button
                      className="add-workout"
                    >
                      Add Workout
                    </button>
                  </Link>,
                ],
                "className": "buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="view-workouts"
                    >
                      Back To Calendar
                    </button>,
                    "replace": false,
                    "to": "/training",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Back To Calendar",
                      "className": "view-workouts",
                    },
                    "ref": null,
                    "rendered": "Back To Calendar",
                    "type": "button",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="add-workout"
                    >
                      Add Workout
                    </button>,
                    "replace": false,
                    "to": "/training/add",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Add Workout",
                      "className": "add-workout",
                    },
                    "ref": null,
                    "rendered": "Add Workout",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no saved workouts",
            "className": "no-data-msg",
          },
          "ref": null,
          "rendered": "You have no saved workouts",
          "type": "h1",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Training All TrainingAll Component should match the snapshot with no user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainingAll
    deleteWorkout={[MockFunction]}
    toggleComplete={[MockFunction]}
    trainingDataUnclean={
      Object {
        "0": Object {
          "completed": true,
          "description": "do stuff",
          "type": "Power",
          "workout_date": "3/3/31",
        },
      }
    }
    user={
      Object {
        "email": "",
        "name": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Log in or sign up to view workouts",
      "className": "no-user-msg",
    },
    "ref": null,
    "rendered": "Log in or sign up to view workouts",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Log in or sign up to view workouts",
        "className": "no-user-msg",
      },
      "ref": null,
      "rendered": "Log in or sign up to view workouts",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
